### ISSUE TITLE
CI Integration (wait for test results in the gradle plugin)

### ISSUE THREAD

**Issue by danielesegato** [awslabs/aws-device-farm-gradle-plugin#15]
The current behavior of the plugin is uploading the artifact on the AWS device farm and then provide an URL to follow it.

For a CI system this is not what you want: you want the test to WAIT to complete and fail if any test has failed / succeed if all test failed.

I'd like to have a task like: `devicefarmUploadAndWait` that makes the gradle script stop and wait for the result (possibly sending multiple modules projects in parallel and wait for all of them).

Without this I can't use AWS Device Farm for my CI.

**Comment by andreschab90** [comment]
It would be awesome, if the gradle task would wait for the test results

**Comment by paulo-raca** [comment]
+1 -- Is anyone working on that?

**Comment by andreschab90** [comment]
i don't think so. i don't need this, because i wrote an own script for my ci server

**Comment by andreas-marschke** [comment]
I got it working including exporting ADF results as JUnit XML. If anyone is interested I can post an MR here.

**Comment by mahyarv** [comment]
@andreas-marschke Can you please post?

**Comment by andreas-marschke** [comment]
@andreschab90 @paulo-raca @danielesegato @mahyarv #22

**Comment by danielesegato** [comment]
@andreas-marschke I've no time to test it right now. But if I did i would appreciate some instruction along with your MR, I guess it may benefit the others as well.

**Comment by andreas-marschke** [comment]
@danielesegato See 

https://github.com/awslabs/aws-device-farm-gradle-plugin/pull/22/files#diff-04c6e90faac2675aa89e2176d2eec7d8R77

For instructions.

Just add this to your `devicefarm` closure to enable waiting
```
useWait()
```

And if you wish to store test results:

```
useOutput("JUnit")
setTestsDestination("build/tests/adf-results")
```

Thats it.

**Comment by erickentry** [comment]
I did try the MR by running 1 test on 3 devices on ADF (1 fail, 2 success). But it seems like the XMLs created were not what I expected... 3 files were created, one of them is:
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite errors="0" failures="0" name="my.app.package.name.testClass" tests="1">
  <testcase classname="test" name="test">
  </testcase>
</testsuite>
```

What did I do wrong?

_This is my first time commenting on GitHub issue, apology if I break any code of conduct_

@andreas-marschke what you did were awesome :+1:

**Comment by andreas-marschke** [comment]
@erickentry Was the test class named test and the test also named test?

If my understanding of how your configuration is supposed to work is correct all 3 runs created their individual run results were stored in these files. Can you attache the files here?

I apologize if my implementation does not support multi-device runs as that was not part of my scope/goal-post when I needed to implement it.

Feel free to file an issue and assign me, worst case. Otherwise the PR is still open, you can review the really simple code that I've written for this to implement it.

**Comment by erickentry** [comment]
@andreas-marschke below were the files generated
[TEST.zip](https://github.com/awslabs/aws-device-farm-gradle-plugin/files/1602171/TEST.zip)

```
Test class name  : ConecdreamChooseCategory
Test method name : test
```

I'll try to implement the XML creation for multi-device runs. If the PR is still open, I'll review. But if it is merged, I'll PR to `master` :)

Thank you so much...

**Comment by igor-brishkoski** [comment]
Any updates here? It would be really helpful instead writing custom scripts.
